cfme/tests/cli/test_appliance_console.py

```diff
@@ -744,27 +745,42 @@ def test_appliance_console_restart():
         testSteps:
             1. 'ap' launches appliance_console.
             2. RETURN clears info screen.
-            3. '16' Stop EVM Server Processes.
-            4. 'Y' Stop CFME Server Gracefully.
-            5. Wait for some time.
-            6. '17' Start EVM Server Processes.
-            7. 'Y' to Start CFME server.
-            8. Wait for some time.
-            9. '21' Quit from appliance_console menu.
-            10. Open appliance IP into Browser.
+            3. '18' restart appliance.
+            4. '1' Restart.
+            5. Verify UI (Open appliance IP into Browser.)
         expectedResults:
             1.
             2.
             3.
             4.
-            5. Cross-check service stopped.
-            6.
-            7.
-            8. Cross-check service started.
-            9.
-            10. Confirm that Appliance is running into browser.
+            5. Confirm that Appliance is running into browser.
     """
-    pass
+    appliance = temp_appliance_preconfig_funcscope
+    old_uptime = appliance.ssh_client.uptime()
+    restart_appliance_cmd = ("ap", RETURN, "18", "1", "Y")
+    appliance.appliance_console.run_commands(restart_appliance_cmd, timeout=30)
+
+    wait_for(
+        net_check, num_sec=180, func_args=["22", appliance.hostname, 2],
+        message='waiting to shutdown machine',
+        fail_condition=True
+    )
+
+    def wait_for_web_ui_up(app_obj, old_uptime):
```

Comment:
Why is this wrapped in a function here? Its just immediately called directly, these statements could just be in-line with the rest of the statements in the test.

If you want to create more complex wait_for loops, you can use its decorator to wrap a set of calls for waiting, but I don't think that's necessary here.

Reply:
Used directly **wait_for_web_ui()** function.